src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -504,9 +504,10 @@ parseNullBinder = underscore *> return NullBinder
 
 parseIdentifierAndBinder :: TokenParser (String, Binder)
 parseIdentifierAndBinder =
-    do name <- lname <|> stringLiteral
+    do name <- lname
        b <- P.option (VarBinder (Ident name)) rest
        return (name, b)
+    <|> (,) <$> stringLiteral <*> rest
```

Comment:
Ohh, right, I got confused... I'm not sure what I was thinking, but I didn't notice that `parseIdentifierAndBinder` is only for record pairs.